src/components/input/Input.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { FC, Dispatch, SetStateAction } from "react";
+import { StyledInput, Label } from "./Input.styled";
+
+export interface Props {
+  value: string;
+  setValue: Dispatch<SetStateAction<string>>;
+  label?: string;
+  placeholder?: string;
+  labelPlacement?: LabelPlacement;
+  width?: string;
+  // Optional arg: If there is some side effect of the change to the input
+  // Pass in this function and it will run when the value changes
+  additionalEffects?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  // Optional arg: If you want to completely control what happens when this input changes
+  // Define this function. Note: value is not set here unless specified in the custom function.
+  customOnChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  onKeyDown?: React.KeyboardEventHandler<HTMLInputElement> | undefined;
+  id?: string;
+}
+
+export type LabelPlacement = "default | overlap";
+
+const TextInput: FC<Props> = ({
+  value,
+  setValue,
```

Comment:
this is misleading, since the caller is not meant to set a value, but expects a change callback here. I would just name this `onChange` or `onValueChange` if you want to denote you are just sending the value. if you need the other callbacks, id name them something like `onPreChange` and `onPostChange` or something that shows an order to the calls

Reply:
Good call.